### PR TITLE
communities: add resource and service handlers for setting default community

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/zenodo/rdm-project/issues/188

### CODE REVIEW

File: invenio_rdm_records/resources/resources.py [hunk 1]
```diff
@@ -368,6 +369,24 @@ def get_suggestions(self):
...
+            id_=resource_requestctx.view_args["pid_value"],
+            identity=g.identity,
+            data=resource_requestctx.data,
+        )
+
+        response = {}
+        if item:
+            response["result"] = item
+        if errors:
+            response["errors"] = errors
```
Comment (jrcastro2) [inline invenio_rdm_records/resources/resources.py:386]: Major: We usually manipulate the response and fill it with errors when we allow to pass multiple values that will be processed and some will success and others might fail. It doesn't seem the case here, we can only set one default community, therefore I would suggest to use error handlers to handle the error. You will need to create a error for [here](https://github.com/inveniosoftware/invenio-communities/pull/1033/files#diff-e062fa3bca790c1a96331a1093f0b6c354d7364099ad90915f96f72d9f046236R170) and then you add the handler for that error [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/config.py#L401).

File: invenio_rdm_records/services/communities/service.py [hunk 2]
```diff
@@ -325,3 +325,26 @@ def search_suggested_communities(
...
+    @unit_of_work()
+    def set_default(self, identity, id_, data, uow):
+        """Set default community."""
+        record = self.record_cls.pid.resolve(id_)
+
+        self.require_permission(identity, "manage", record=record)
+
+        errors = []
+        try:
+            record.parent.communities.default = data["data"]
```
Comment (jrcastro2) [inline invenio_rdm_records/services/communities/service.py:338]: Major: Usually when processing data, we do use marshmallow schemas to validate it before processing.

File: invenio_rdm_records/services/communities/service.py [hunk 3]
```diff
@@ -325,3 +325,26 @@ def search_suggested_communities(
...
+        except AttributeError as ex:
+            errors.append({"message": str(ex)})
```
Comment (jrcastro2) [inline invenio_rdm_records/services/communities/service.py:340]: Major: Connected to the comment on the resource layer, the errors we do only want to control if we are processing multiple operations, whichc doesn't seem the case here, I would just raise the exception and let the error_handler take care of it. Meaning that there is no need to control the errors here

File: invenio_rdm_records/services/config.py [hunk 4]
```diff
@@ -179,6 +180,7 @@ class RDMRecordCommunitiesConfig(ServiceConfig, ConfiguratorMixin):
     )
 
     schema = RecordCommunitiesSchema
+    communities_schema = CommunitiesSchema
```
Comment (rekt-hard) [inline invenio_rdm_records/services/config.py:183]: Minor: `CommunitiesSchema` may also benefit from adding a unique and duplicate id check as in https://github.com/anikachurilova/invenio-rdm-records/blob/22fe2376431cfd84047c33a9b05dd6f5624d3cde/invenio_rdm_records/services/schemas/record_communities.py#L30-L55

File: invenio_rdm_records/services/communities/service.py [hunk 5]
```diff
@@ -325,3 +330,27 @@ def search_suggested_communities(
...
+            data,
+            context={
+                "identity": identity,
+            },
+            raise_errors=True,
+        )
+
+        record = self.record_cls.pid.resolve(id_)
+        self.require_permission(identity, "manage", record=record)
+        record.parent.communities.default = valid_data["default"]
```
Comment (rekt-hard) [inline invenio_rdm_records/services/communities/service.py:347]: minor: should we check if a community with the default id exists? Maybe even check if all passed ids exist?

**Review by rekt-hard — COMMENTED** [review]
Minor input on value validation check for ids.

File: invenio_rdm_records/services/communities/service.py (hunk 5 above)
Comment (rekt-hard) [reply invenio_rdm_records/services/communities/service.py:347]: Sorry, just saw the other PR where the check happens :)

**Pull request merged**